internal/database/lock.go

```diff
@@ -89,6 +91,36 @@ func makeMultiUnlockFn(ctx context.Context, db *DB, lockIDs []string, expires ti
 	}
 }
 
+func serializationError(err error) bool {
+	// See https://www.postgresql.org/docs/current/errcodes-appendix.html
+	if pgErr, ok := err.(*pgconn.PgError); !ok || pgErr.Code == "40001" {
+		return true
+	}
+	return false
+}
+
+// LockRetry will attempt to acquire a lock using exponential backoff, up to the tryFor time.
+// Delegates to db.Lock().
+func (db *DB) LockRetry(ctx context.Context, lockID string, ttl time.Duration, tryFor time.Duration) (UnlockFn, error) {
+	// Ignored error - it checks for < 0
+	b, _ := retry.NewExponential(time.Millisecond)
+	b = retry.WithMaxDuration(tryFor, b)
+
+	var unlock UnlockFn
+	err := retry.Do(ctx, b, func(ctx context.Context) error {
+		var err error
+		unlock, err = db.Lock(ctx, lockID, ttl)
+		if err != nil && serializationError(err) {
```

Comment:
If you change isSerializationError to return false when err is nil, you can simplify this and future calls